core/src/services/s3/backend.rs

```diff
@@ -1026,7 +1062,7 @@ impl Accessor for S3Backend {
 
     async fn create(&self, path: &str, _: OpCreate) -> Result<RpCreate> {
         let mut req =
-            self.s3_put_object_request(path, Some(0), None, None, None, AsyncBody::Empty)?;
+            self.s3_put_object_request(path, Some(0), None, None, None, None, AsyncBody::Empty)?;
```

Comment:
default_storage class stored at `Backend`. So we don't need to pass it via request.

Reply:
🤔 so the `storage_class` can only be set upon the creation of S3 backend and is immutable afterwards, right? As always, I tend to make the API flexible as storage class can be configured at the object level, but let's KISS for now.

Reply:
> so the `storage_class` can only be set upon the creation of S3 backend and is immutable afterwards, right?

Yep

> As always, I tend to make the API flexible as storage class can be configured at the object level, but let's KISS for now.

Unfortunately, as I mentioned earlier, this feature goes against our VISION. Therefore, we cannot support it.

Reply:
Got it, please take another look